library/rsa.c

```diff
@@ -1628,17 +1628,17 @@ static int rsa_rsassa_pkcs1_v15_encode( mbedtls_md_type_t md_alg,
      *                 TAG-OCTET + LEN [ HASH ] ]
      */
     *p++ = MBEDTLS_ASN1_SEQUENCE | MBEDTLS_ASN1_CONSTRUCTED;
-    *p++ = 0x08 + oid_size + hashlen;
+    *p++ = (unsigned char)( 0x08 + oid_size + hashlen );
     *p++ = MBEDTLS_ASN1_SEQUENCE | MBEDTLS_ASN1_CONSTRUCTED;
-    *p++ = 0x04 + oid_size;
+    *p++ = (unsigned char)( 0x04 + oid_size );
     *p++ = MBEDTLS_ASN1_OID;
-    *p++ = oid_size;
+    *p++ = (unsigned char) oid_size;
     memcpy( p, oid, oid_size );
     p += oid_size;
     *p++ = MBEDTLS_ASN1_NULL;
     *p++ = 0x00;
     *p++ = MBEDTLS_ASN1_OCTET_STRING;
-    *p++ = hashlen;
+    *p++ = (unsigned char) hashlen;
```

Comment:
Consider additionally masking (`*p++ = x & 0xFF`) to avoid warnings about loss of precision in some compilers. @dgreen-arm has been working on eliminating integer conversion warnings on a number of platforms and could comment further here.

Reply:
@Patater Are there compilers that would warn on an _explicit_ truncation cast? I was expecting this to happen only for implicit downcasts. @dgreen-arm Please comment.

Reply:
I can't speak for all compilers, but I've been working on the conversion warnings with gcc. Making this change removes the warning that was present here before, at least with gcc.

Reply:
@dgreen-arm Thanks! 

I'd find it strange if the only way to ease a compiler would be to do a masking. Imho, making the cast explicit is a cleaner way of saying "I know what I'm doing, I want this truncation".

Reply:
Looks like this is only fiddly and requires a mask in gcc when assigning to bitfields.

Toy example at https://godbolt.org/g/HzYSFS

Reply:
@Patater Thanks, that makes sense!